launcher/minecraft/mod/tasks/ModFolderLoadTask.cpp

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2024
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "ModFolderLoadTask.h"
+
+#include "Application.h"
+#include "FileSystem.h"
+#include "minecraft/mod/MetadataHandler.h"
+
+#include <QThread>
+
+ModFolderLoadTask::ModFolderLoadTask(const QDir& root_dir,
+                                     bool is_indexed,
+                                     bool clean_orphan,
+                                     std::function<Resource*(const QFileInfo&)> create_function)
+    : Task(false)
+    , m_root_dir(root_dir)
+    , m_is_indexed(is_indexed)
+    , m_clean_orphan(clean_orphan)
+    , m_create_func(create_function)
+    , m_result(new Result())
+    , m_thread_to_spawn_into(thread())
+{}
+
+void ModFolderLoadTask::executeTask()
+{
+    if (thread() != m_thread_to_spawn_into)
+        connect(this, &Task::finished, this->thread(), &QThread::quit);
```

Comment:
```suggestion
    if (thread() != m_thread_to_spawn_into) {
        connect(this, &Task::finished, this->thread(), &QThread::quit);
    }
```
Please always use braces for conditional or loop blocks. unless you putting them on one line like
```cpp
if (test) one_line_expression.op();
```
It's much easier to read
also, please run clang-format and clang-tidy. the code is wildly out of style with the rest of the codebase.

Reply:
while the above lines are addressed the pattern still appears more than 10 times in other files in the commits